src/dmd/expression.h

```diff
@@ -231,6 +231,7 @@ class IntegerExp : public Expression
 {
 public:
     dinteger_t value;
+    bool literealFlag;
```

Comment:
Little typo. I think `isLiteral` would be clearer (although 'literal' itself might be considered confusing, as all `IntegerExp` are AST literals - `isReadonlyLiteral` might be another option).

I've just grepped for `setInteger`, and it looks like you replaced the only occurrence in `expressionsem.d`...

Reply:
Yes, almost a good reason to drop the flag if there was not the Expression.type, which is a public field